resources/idlharness.js

```diff
@@ -3288,15 +3292,24 @@ function idl_test(srcs, deps, idl_setup_func) {
         deps = (deps instanceof Array) ? deps : [deps] || [];
         var setup_error = null;
         return Promise.all(
-            srcs.concat(deps).map(function(spec) {
-                return fetch_spec(spec);
-            }))
-            .then(function(idls) {
+            srcs.concat(deps).map(fetch_spec))
+            .then(function(results) {
+                const astArray = results.map(result =>
+                    WebIDL2.parse(result.idl, { sourceName: result.spec })
+                );
+                const validations = WebIDL2.validate(astArray);
+                if (validations.length) {
+                    const message = validations.join("\n\n");
+                    throw new Error(message);
+                }
+                return astArray;
+            })
+            .then(function(astArray) {
```

Comment:
Is there a reason to add this `then()`? I think you can just continue from the previous handler.

Reply:
Oops, it's a mistake. Will remove it.